### PR TITLE
fix(dygraphs): Allow nullable `valueRange` array type

### PR DESCRIPTION
**Pull request by mdtusz**
When setting a `valueRange` option on a dygraph chart, it is valid to provide a `null` value for the range, in which case the chart range will auto-detect the upper or lower bound.

Doc reference: http://dygraphs.com/options.html#valueRange

This change modifies the required option type to accept an array with `null` elements, as well as constrains the value to be a tuple array of length 2.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: http://dygraphs.com/options.html#valueRange
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## dygraphs ([<kbd>unpkg</kbd>](https://unpkg.com/browse/dygraphs@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'dygraphs'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
1. NAME
1. VERSION
1. toString
1. DEFAULT_ROLL_PERIOD

<details>
<summary>as well as these 24 other properties...</summary>
<p>DEFAULT_WIDTH, DEFAULT_HEIGHT, ANIMATION_STEPS, ANIMATION_DURATION, KMB_LABELS, KMG2_BIG_LABELS, KMG2_SMALL_LABELS, numberValueFormatter, numberAxisLabelFormatter, SHORT_MONTH_NAMES_, dateAxisLabelFormatter, dateAxisFormatter, dateValueFormatter, Plotters, DEFAULT_ATTRS, HORIZONTAL, VERTICAL, PLUGINS, addedAnnotationCSS, zoomAnimationFunction, PointType, stackPoints_, mapLegacyOptions_, addAnnotationRule</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 98cd699c4a246219665ac143f7875bd9d7316220
</p>

File: types/dygraphs/index.d.ts [hunk 1]
```diff
@@ -252,7 +252,7 @@ declare namespace dygraphs {
          * be calculated automatically (e.g. [null, 30] to automatically calculate just the lower
          * bound)
          */
-        valueRange?: number[];
+        valueRange?: [number | null, number | null];
```
Comment (danvk) [inline types/dygraphs/index.d.ts:255]: It looks like `null` (not an array) is also allowed, to zoom to the defaults:

```ts
g.updateOption({ valueRange: null });
```

**Review by danvk — CHANGES_REQUESTED** [review]
Overall looks good but I agree with TypeScript Bot — please add a test to [`dygraphs-tests.ts`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/98cd699c4a246219665ac143f7875bd9d7316220/types/dygraphs/dygraphs-tests.ts) showing expected usage.

**Comment by mdtusz** [comment]
I'm not 100% clear on the mechanism for testing - it seems like the test files are just typechecked? 

Hope what has been added is ok, but am happy to add or change it up more if required.

**Review by danvk — APPROVED** [review]
Thanks @mdtusz, this looks great.

Those tests are fine for the time being. Testing type declarations is an interesting & subtle topic — I gave a whole [talk about it](https://www.youtube.com/watch?v=nygcFEwOG8w) at tsconf last year if you're interested in the details.

**Comment by mdtusz** [comment]
Ready to merge

**Pull request merged**